sysutils/pfSense-pkg-LCDproc/files/usr/local/pkg/lcdproc_client.php

```diff
@@ -1339,6 +1390,52 @@ function loop_status($lcd) {
 					$lcd_cmds[] = "widget_set {$name} title_wdgt 1 1 \"{$title}\"";
 					$lcd_cmds[] = "widget_set {$name} text_wdgt 1 2 {$lcdpanel_width} 2 h 4 \"{$cputemperature}\"";
 					break;
+				case "scr_ntp":
+					$ntp_hms = date("H:i");
+					$ntpq_output = lcdproc_get_ntp_status();
+					if ($ntpq_output != false) {
+						$ntptime = 
+						$ntpq_counter = 0;
+						foreach ($ntpq_output as $line) {
+							if (substr($line, 0, 1) == "*" || substr($line, 0, 1) == "o") {
+								//Active NTP Peer
+								$line = substr($line, 1);
+								$peerinfo = preg_split("/[\s\t]+/", $line);
+								$peerdel = substr($peerinfo[7],0,5);
+								$peeroff = substr($peerinfo[8],0,6);
+								$peerjit = substr($peerinfo[9],0,5);
+								if ($peerinfo[2] == "0" && $peerinfo[1] == ".GPS.") {
+									// When local GPS is used. Show the IP of the source and the stratum. Also append how many satellites are in use or in view
+									$gps_sat_count = lcdproc_get_ntp_gps_sat_count();
+									$lcd_cmds[] = "widget_set {$name} time_st_wdgt 1 1 {$lcdpanel_width} 2 h 4 \"NTP: {$ntp_hms} Strtm: {$peerinfo[2]}\"";
+									$lcd_cmds[] = "widget_set {$name} ref_wdgt 1 2 {$lcdpanel_width} 2 h 4 \"Ref: {$peerinfo[1]} " . (($gps_sat_count == false) ? "" : "{$gps_sat_count}") . "\"";
```

Comment:
This is a bit confusing but it may be the comment that is wrong and not the code.

The comment above says it should be showing the IP address but that would be in `$peerinfo[0]`. As-is here it would just be printing `.GPS.` due to the `if` test, which seems correct to me. When the reference is a local GPS, the "IP address" is bunk, just an internal signal to ntpd that it should be using the local GPS.

It seems like in the case of a local GPS, though, printing `.GPS.` like you are already, or maybe printing a string saying "Local GPS Device", is best.

tl;dr: If the code is doing what you want, adjust the comment to match.

Reply:
I'll update the comment.  

I was trying to avoid a long description when on a local GPS to leave enough room for the number of satellites in-view or in-use (depending on which NMEA messages are transmitted) and prevent row scrolling.  

P.S. I think I finally figured out I can submit updates without having to generate a new pull request.

Reply:
Submitted updated lcdproc_client.php